### DISCUSSION TITLE
Vercel Site not indexing

### DISCUSSION THREAD

**Discussion by caycewilliams** [vercel/community#6723] [Help]
### Summary

I can't index my site with google because the vercel version is being found. Is there a way to prevent robots on this vercel site or prevent indexing?

See error below from google:
The URL selected by Google as the authoritative version of this page. Other versions can be served in search results, depending on factors such as the user's device type or language. This is not available in the live test, as Google selects a canonical URL only after a page is indexed.
<img width="794" alt="Screenshot 2024-05-03 at 11 00 42 AM" src="https://github.com/vercel/community/assets/6610028/c42fe57b-7b19-483d-b857-b79cc0d32bb7">

### Example

https://fullwebsitescreenshot.com

### Steps to Reproduce

Host a new vercel site and try submitting the home page for indexing with a canonical set to the same page. Go to google search console and see if it is being indexed.

**Comment by versecafe** [comment]
@caycewilliams you can use a `robots.ts` file instead and do conditional indexing where it only allows if not the Vercel domain, you also should have your `sitemap.xml`/`sitemap.ts` point explicitly to your production URL like this

```ts
return [
    {
      url: "https://fullwebsitescreenshot.com",
      lastModified: new Date(),
      changeFrequency: "monthly",
      priority: 1,
    },
]
 ```

**Comment by caycewilliams** [comment]
Hey thanks for your response @versecafe! I have a sitemap, but I think since vercel makes a copy, it replicates. I've submitted the sitemap with the primary url to google search console.

Do you have example of robots.txt?

I've also put in code to add the noindex tag on the vercel domain only, hoping that works.


```
export async function generateMetadata({
  params,
}: {
  params: { slug: string };
}) {
  let url = headers().get("x-url");
  const isVercel = url?.includes('vercel.app')
  const meta = {
    title: 'Full Website Screenshot 📸',
    description: 'Enter any url and get a full website screenshot.',
    
  } as Metadata
  if (isVercel) meta.robots = 'noindex'
  return meta
  })
```
